tests/common.go

```diff
@@ -5311,3 +5318,122 @@ func IsEksPxOperator() bool {
 	}
 	return false
 }
+
+// ConcurrentExecute executes the given operation on each item in the collection, either sequentially
+// or in parallel, depending on the specified execution mode. It also returns an error when items is not
+// of type slice, array or map.
+//
+// Parameters:
+//
+//	items: The collection of items to operate on (slice, array, or map).
+//	operation: The function to execute on each item. If the function takes one argument,
+//	           it will be passed the item value. If it takes two arguments, the first
+//	           will be the item key or index, and the second will be the item value.
+//	executionMode: The mode to use for executing the operation, either "Sequential" or "Parallel".
+//
+// # Example
+//
+// The original code:
+//
+//	for _, value := range items /*slice, array or map*/ {
+//		operation(value)
+//	}
+//
+// or
+//
+//	for index, value := range items /*slice or array*/ {
+//		operation(index, value)
+//	}
+//
+// or
+//
+//	for key, value := range items /*map*/ {
+//		operation(key, value)
+//	}
+//
+// The original code uses a common pattern for iterating over a slice, array or map of items and calling the 'operation'
+// function for each item. To simplify this pattern and allow for concurrent execution of the 'operation'
+// function you can replace the for loops with a call to ConcurrentExecute(items, operation, executionMode), where
+// 'executionMode' is either 'Parallel' or 'Sequential'.
+func ConcurrentExecute(items interface{}, operation interface{}, executionMode ExecutionMode) error {
```

Comment:
Rename this function

Reply:
Discarded the function because it was not needed in this test instance.